e2e/LongHaul/device/IotHub.cs

```diff
@@ -198,22 +161,26 @@ public async Task RunAsync(CancellationToken ct)
             _messagesToSend.Enqueue(iotMessage);
         }
 
-        public async Task SetPropertiesAsync(object properties, CancellationToken cancellationToken)
+        public async Task SetPropertiesAsync(SystemProperties properties, CancellationToken cancellationToken)
         {
             Debug.Assert(_deviceClient != null);
             Debug.Assert(properties != null);
 
-            string propertiesPayload = JsonSerializer.Serialize(properties, _jsonSerializerOptions);
-            Debug.Assert(!string.IsNullOrWhiteSpace(propertiesPayload));
+            var reportedProperties = new ReportedProperties
+            {
+                { "systemArchitecture", properties.SystemArchitecture },
```

Comment:
Ah, this is something @abhipsaMisra hasn't added support for yet. In our current API, one can only set an object if they have a key for it. I want to be able to set a payload with a single parameter that has all the key/value pairs we want to set.

Anyway, we shouldn't have to hardcode these keys and this method shouldn't be hardcoded to only `SystemProperties`. Let's talk and work out how we enable this.